src/command_modules/azure-cli-vm/azure/cli/command_modules/vm/_params.py

```diff
@@ -140,6 +141,9 @@ def get_vm_size_completion_list(prefix, action, parsed_args, **kwargs):  # pylin
 register_cli_argument('vmss extension image', 'top', help='Return top number of records')
 register_cli_argument('vmss extension image', 'version', help='Extension version')
 
+register_cli_argument('vmss extension', 'settings', type=validate_file_or_dict)
```

Comment:
The help text will still say JSON, and that is what is encouraged. However, for the CMD users (like moi) who need to work around this issue, they could also use the nearly identical dictionary syntax.